spec/controllers/menus_controller_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'rails_helper'
+
+RSpec.describe MenusController, type: :controller do
+  render_views
+
+  let!(:user) { create(:user) }
+  let(:product_1) { create(:product, name: 'Product 1') }
+  let(:product_2) { create(:product, name: 'Product 2') }
+  let(:product_3) { create(:product, name: 'Product 3') }
+
+  let!(:menu_1) { create(:menu, product_ids: [product_1.id, product_2.id]) }
+  let!(:menu_2) { create(:menu, product_ids: [product_3.id]) }
+
+  before { sign_in user }
+    
+  describe 'GET index' do
+    before { get :index }
+
+    specify do
+      expect(response).to have_http_status(200) 
+      expect(response).to render_template(:index) 
+      expect(assigns(:menus)).to eq [menu_1, menu_2]
+    end
+  end
+
+  describe 'GET new' do
+    before { get :new }
+
+    specify do
+      expect(response).to have_http_status(200) 
+      expect(response).to render_template(:new)
+    end
+  end 
+
+  describe 'POST create' do
+
```

Comment:
should create and login with other user 
`    let!(:user) { create(:user, email: 'test@test.com') }
`

Reply:
tạo 1 user mới để sign in, sau đó dùng để kiểm tra created_by